src/axom/mint/docs/sphinx/sections/execution_model.rst

```diff
@@ -0,0 +1,178 @@
+  :ref:`sections/execution_model`. It defines the body of operations
+  performed on the supplied mesh.
+
+See the :ref:`sections/tutorial` for code snippets that illustrate how to use
+the :ref:`NodeTraversalFunctions`, :ref:`CellTraversalFunctions` and
+:ref:`FaceTraversalFunctions` of the :ref:`sections/execution_model`.
+
+.. _executionPolicy:
+
+Execution Policy
+^^^^^^^^^^^^^^^^^
+
+The :ref:`ExecutionPolicy` is specifed as the first template argument and is
+required by all of the constituent functions of the
+:ref:`sections/execution_model`. Mint defines a set of high-level execution
+policies, summarized in the table below.
+
++-------------------------+------------------------+---------------------------------+
+| Execution Policy        |     Requirements       |          Description            |
+|                         |                        |                                 |
++=========================+========================+=================================+
+| **serial**              |    None.               | Serial execution on             |
+|                         |                        | the CPU.                        |
++-------------------------+------------------------+---------------------------------+
+| **parallel_cpu**        |  `RAJA`_ + OpenMP      | Parallel execution on           |
+|                         |                        | the CPU with OpenMP.            |
++-------------------------+------------------------+---------------------------------+
+| **parallel_gpu**        |  `RAJA`_ + CUDA        | Parallel execution on           |
+|                         |                        | CUDA-enabled GPUs.              |
++-------------------------+------------------------+---------------------------------+
+| **parallel_gpu_async**  |  `RAJA`_ + CUDA        | Asynchronous parallel           |
+|                         |                        | execution on CUDA-enabled GPUs. |
++-------------------------+------------------------+---------------------------------+
+
+These policies are mapped to corresponding `RAJA`_ execution policies internally.
+
+.. note::
+
+   Mint's execution policies are encapsulated in the ``axom::mint::policy::``
+   namespace.
+
+.. _executionSignature:
+
+Execution Signature
+^^^^^^^^^^^^^^^^^^^^
+
+The :ref:`executionSignature` is specified as the second, *optional* template
+argument to the constituent functions of the :ref:`sections/execution_model`.
+The :ref:`executionSignature` indicates the list of arguments that are
+supplied to the user-specified kernel.
+
+.. note::
+
+    If not specified, the default :ref:`executionSignature` to is set to
+    ``mint::xargs::index``, which, indicates that the supplied kernel takes
+    a single argument that corresponds to the index of the corresponding
+    iteration space, i.e, the loop index.
+
+The list of available :ref:`executionSignature` options is summarized below:
```

Comment:
- Are these the only execution signatures ever needed?  
- Why did you choose this set of signatures? 
- Is there a reference or recommendation, ideally such as the link to the CGNS node orderings?
- Is it easy (or possible) to add an execution signature, should a code ever need a new one?
- What happens if you get it wrong, such as give `mint::xargs::cellids` to a Cell Traversal Function?

Reply:
* We can extend the execution signature to add more in the future. These are not all that we would ever need. But, it is straightforward to extend

* The execution signatures are selected based on things that I know I need now and in the near future. 

* I am not sure what you mean by the 3rd bullet. Recommendation for the execution signatures?

* Yes, it is straightforward to add a new signature. It is meant to be extensible. 

* Since these are template arguments that dispatch to runtime calls, if you pass something that is not supported you will get a compiler error. I think I've been good with using static assert to provide meaningful messages, but, maybe there is room to grow in that regard.

Reply:
- Bullets 1, 2, 4: Thanks for the clarification.  I recommend making these points in the text, perhaps something like  "... summarized below.  We developed these signatures based on the usage patterns found in LLNL physics codes.  By design, it is straightforward to add a new execution signature when needed."
- Bullet 3: If there were a survey paper or well known reference discussing access patterns, here would be one place to put it.  As a library user, I'd like to see a reason behind the selection of stuff (here, execution signatures) offered.  If you use the sentence I suggested, it explains why Mint offers these signatures.
- Bullet 5: Please include this fact in the docs.  I'd recommend to add something like the following sentence either to the Note or the paragraph preceding it:  "If a call through the mint execution model specifies an execution signature that does not match the traversal function, a compile time error will result."

Reply:
Done.